### PR TITLE
Remove react-test-renderer

### PR DESCRIPTION
**Pull request by smoralesd**
#### Description of changes

Remove usage of react-test-renderer in favor of enzyme (shallow, mount, render...).
Remove ShallowRender & ShallowRenderer.
Migrate every test using the previously mention lib/classes to use enzyme.
Remove react-test-render from our dependencies.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`npm run test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`npm run precheckin`)
- [ ] Added screenshots/GIFs for UI changes.

### CODE REVIEW

File: src/tests/unit/tests/DetailsView/reports/components/assessment-report-step-list.test.tsx [hunk 1]
```diff
@@ -23,8 +22,8 @@ describe('AssessmentReportStepListTest', () => {
             steps: AssessmentReportBuilderTestHelper.getRequirementReportModelPass(),
         };
 
-        const shallow = shallowRender(<AssessmentReportStepList {...props} />);
-        expect(shallow).toMatchSnapshot(); //
+        const wrapper = shallow(<AssessmentReportStepList {...props} />);
+        expect(wrapper.getElement()).toMatchSnapshot(); //
```
Comment (waabid) [inline src/tests/unit/tests/DetailsView/reports/components/assessment-report-step-list.test.tsx]: Any reason for this '//' here (and in the file)?

Comment (smoralesd) [reply src/tests/unit/tests/DetailsView/reports/components/assessment-report-step-list.test.tsx]: :( not really... didn't even see it really.

Comment (smoralesd) [reply src/tests/unit/tests/DetailsView/reports/components/assessment-report-step-list.test.tsx]: updated

File: src/tests/unit/tests/DetailsView/reports/components/assessment-report-step-list.test.tsx [hunk 2]
```diff
@@ -36,8 +35,8 @@ describe('AssessmentReportStepListTest', () => {
             steps,
         };
 
-        const shallow = shallowRender(<AssessmentReportStepList {...props} />);
-        expect(shallow).toMatchSnapshot(); //
+        const wrapper = shallow(<AssessmentReportStepList {...props} />);
+        expect(wrapper.getElement()).toMatchSnapshot(); //
```
Comment (peterdur) [inline src/tests/unit/tests/DetailsView/reports/components/assessment-report-step-list.test.tsx]: Extra empty comments are still here (and at lines 50, 61).

Comment (peterdur) [inline src/tests/unit/tests/DetailsView/reports/components/assessment-report-step-list.test.tsx]: Extra comment
```suggestion
        expect(wrapper.getElement()).toMatchSnapshot();
```

File: src/tests/unit/tests/DetailsView/reports/components/assessment-report-step-list.test.tsx [hunk 3]
```diff
@@ -47,8 +46,8 @@ describe('AssessmentReportStepListTest', () => {
             steps: AssessmentReportBuilderTestHelper.getRequirementReportModelFail(),
         };
 
-        const shallow = shallowRender(<AssessmentReportStepList {...props} />);
-        expect(shallow).toMatchSnapshot();
+        const wrapper = shallow(<AssessmentReportStepList {...props} />);
+        expect(wrapper.getElement()).toMatchSnapshot(); //
```
Comment (peterdur) [inline src/tests/unit/tests/DetailsView/reports/components/assessment-report-step-list.test.tsx]:
```suggestion
        expect(wrapper.getElement()).toMatchSnapshot();
```

File: src/tests/unit/tests/DetailsView/reports/components/assessment-report-step-list.test.tsx [hunk 4]
```diff
@@ -58,7 +57,7 @@ describe('AssessmentReportStepListTest', () => {
             steps: AssessmentReportBuilderTestHelper.getRequirementReportModelUnknownStep3(),
         };
 
-        const shallow = shallowRender(<AssessmentReportStepList {...props} />);
-        expect(shallow).toMatchSnapshot();
+        const wrapper = shallow(<AssessmentReportStepList {...props} />);
+        expect(wrapper.getElement()).toMatchSnapshot(); //
```
Comment (peterdur) [inline src/tests/unit/tests/DetailsView/reports/components/assessment-report-step-list.test.tsx]:
```suggestion
        expect(wrapper.getElement()).toMatchSnapshot();
```

File: src/tests/unit/tests/DetailsView/reports/components/assessment-report-step-header.test.tsx [hunk 5]
```diff
@@ -1,8 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-import * as Enzyme from 'enzyme';
+import { shallow } from 'enzyme';
```
Comment (Shobhit1) [inline src/tests/unit/tests/DetailsView/reports/components/assessment-report-step-header.test.tsx]: 👍

**Pull request merged**